src/canvas.class.ts

```diff
@@ -441,10 +441,12 @@ import { saveObjectTransform } from './util/misc/objectTransforms';
         this._objectsToRender = undefined;
         // removing active object should fire "selection:cleared" events
         if (obj === this._activeObject) {
-          this.fire('before:selection:cleared', { target: obj });
+          this.fire('before:selection:cleared', { deselected: [obj] });
```

Comment:
So here i m thorn between deselected and maybeDeselected or something like that.
Typescript allow for self descovery of data structure, that means that we can care a bit less about consistency ( all events have target! ) and have names that make sense.
In this case deselected is past, while the before:selection:cleared can be interrupted and is not deselected yet.

Ideas?

Reply:
I get your point. A good point indeed

Reply:
MaybeDeselcted is bad naming. Target or desecltionTarget are better